src/server.h

```diff
@@ -2159,7 +2159,7 @@ typedef int redisGetKeysProc(struct redisCommand *cmd, robj **argv, int argc, ge
  */
 struct redisCommand {
     /* Declarative data */
-    const char *name; /* A string representing the command name. */
+    const char *fullname; /* A string representing the command fullname. */
```

Comment:
we have an odd situation where non-subcommand have `const char *` and subcommand have `sds`
I suggest changing `fullname` to be `sds` (and call `sdsnew` populateCommandStructure)

this will also help in removing some of the implicit casting done (mostly in module.c)
also, we can change some `sdsnew` to `sdsdup`